src/Chainweb/Version.hs

```diff
@@ -637,6 +637,29 @@ pactBackCompat_v16 Mainnet01 h = h < 328000
 pactBackCompat_v16 Development h = h < 120
 pactBackCompat_v16 _ _ = False
 
+-- | If this is true the creation time of the current header is used for pact tx
```

Comment:
I'm concerned that it seems we're introducing this complexity for `Development` only. Mainnet is always `True`, and other versions are basically duplicative of switches on `isGenesis`. Wouldn't it make more sense to have a special case for "first non-genesis block" to use the current header time?

Reply:
We are going to need the switch for Mainnet. We'll probably use the value for `Development` for testing. All other test version it does't really matter, as long as tests pass.

As we did with other guards, we might be able to remove it after the transition is completed and replace it by a hard-coded timing validation grace period, which would allow to also remove related code logic.

Reply:
I have an upcoming PR, that makes the timing behavior in several tests deterministic (avoiding use if getCurrentTime). With that change (possibly extending it to more test) it may be possible to remove the special case for test versions.